File: HW3/driver2.cpp
#include "Tetro.h"

using namespace GTU_hamza;


Tetrominos * Tetrominos::all_shapes;
int Tetrominos::size_of_tetrominos=0;
int main()
{
    char *in=nullptr;
    char *temp=nullptr;
   
    cout<<"========================= WELCOME TETRO GAME ====================="<<endl;
    cout<<"========================= DRIVER1 HAS RUNNED. GAME WILL PLAY MANUALLY.\nENTER THE TETROMİNOS NAMES AFTER THE ENTER BOARD SIZE  ====================="<<endl;
    cout<<"Please determine the size of game board(like 3x5(row is 3 and colum is 5) ";
    in=Tetro::get_line(in);
    Tetro a(in);
    //cout<<"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"<<endl;
    //a.Play_game();
    //cout<<"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"<<endl;

    cout<<"Enter Terminos Name "<<endl;
    Tetrominos *obj=nullptr;
    while(true)   // cin>>Tetro::char_cmpr(in,   ) 
    {   
        in=nullptr;
        cout<<"Enter Tetromino Name Please!!"<<endl;
        in=Tetro::get_line(in);
       
        if(Tetro::char_cmpr(in,"I")||Tetro::char_cmpr(in,"i"))
        {
            obj= new Tetrominos(shapes::I);
            Tetrominos::set_shape(*obj);
        }
        else if(Tetro::char_cmpr(in,"O")||Tetro::char_cmpr(in,"o"))
        {
            obj= new Tetrominos(shapes::O);
            Tetrominos::set_shape(*obj);
           
        }
        else if(Tetro::char_cmpr(in,"T")||Tetro::char_cmpr(in,"t"))
        {
            obj= new Tetrominos(shapes::T);
            Tetrominos::set_shape(*obj);
           
        }
        else if(Tetro::char_cmpr(in,"J")||Tetro::char_cmpr(in,"j"))
        {
            obj= new Tetrominos(shapes::J);
            Tetrominos::set_shape(*obj);
           
        }
        else if(Tetro::char_cmpr(in,"S")||Tetro::char_cmpr(in,"s"))
        {
            obj= new Tetrominos(shapes::S);
            Tetrominos::set_shape(*obj);
           
        }
        else if(Tetro::char_cmpr(in,"Z")||Tetro::char_cmpr(in,"z"))
        {
            obj= new Tetrominos(shapes::Z);
            Tetrominos::set_shape(*obj);
           
        }
        else if(Tetro::char_cmpr(in,"L")||Tetro::char_cmpr(in,"l"))
        {
            obj= new Tetrominos(shapes::L);
            Tetrominos::set_shape(*obj);
           
        }
        else if(Tetro::char_cmpr(in,"Q")||Tetro::char_cmpr(in,"q"))
        {
            cerr<<"You have been quit game.Game will be terminated "<<endl;
            break;
            
        }
        else if(Tetro::char_cmpr(in,"R")||Tetro::char_cmpr(in,"r"))
        {
            obj= new Tetrominos(shapes::O);
            Tetrominos::set_shape(*obj);
            // Tetrominos::set_shape(obj);
           break;
        }
        else
            cout<<"Invalid tetrominos trying !!!Try Again!!"<<endl;
        if(obj!=nullptr)
        {

        int move_count,move_direct,direct_count,rotate_direct;
        a+= Tetrominos::all_shapes[Tetrominos::get_size_of_tetrominos()-1];  
        cout<<"Choose Rotate Direction \n"
        <<"1-LEFT\n"
        <<"2-RIGHT\n"
        <<"3⁻NONE"<<endl;
        while(true)
        {
            temp=Tetro::get_line(temp);
            if((temp[0]<'1'||temp[0]>'3')||Tetro::str_len(temp)>1||Tetro::str_len(temp)==0) 
            {
                cout<<"Invalid input.try Again!!"<<endl;
                delete []temp;
                temp=nullptr;
            } 
            else
                break;

        }
        
        rotate_direct=stoi(temp);
        delete [] temp;
        temp=nullptr;
        
        while(true)
        {
            cout<<"Enter count of rotate  direction number\n";
            temp=Tetro::get_line(temp);
            if((temp[0]<'1'||temp[0]>'5')||Tetro::str_len(temp)>1||Tetro::str_len(temp)==0) 
            {
                cout<<"Invalid input.try Again!!"<<endl;
                delete []temp;
                temp=nullptr;
                //return Play_game();
            }
            else 
                break;
        }
       
        direct_count=stoi(temp);
        delete [] temp;
        temp=nullptr;
        cout<<"Choose Move Direction \n"
        <<"1-LEFT\n"
        <<"2-RIGHT\n"
        <<"3⁻NONE"<<endl;
        
        while (true)
        {
            temp=Tetro::get_line(temp);
            if((temp[0]<'1'||temp[0]>'3')||Tetro::str_len(temp)>1||Tetro::str_len(temp)==0) 
            {
                cout<<"Invalid input.try Again!!"<<endl;
                delete []temp;
                temp =nullptr;
            }
            else
                break;
            /* code */
        }
        
        move_direct=stoi(temp);
    
        delete [] temp;
        temp=nullptr;
        
        while (true)
        {
            cout<<"Enter count of move  direction number\n";
            temp=Tetro::get_line(temp);
            if(a.is_number(temp)) 
            {
                move_count=stoi(temp);
                delete [] temp;
                temp=nullptr;
                break;
            }
            else
                cout<<"Invalid input.try Again!!"<<endl;
            /* code */
        }
                a.Play_game(move_count,move_direct,direct_count,rotate_direct);    
        }
        delete [] in;
         
    }


    //std::this_thread::sleep_for(std::chrono::milliseconds(50));
    //a.Draw();

}

File: HW7-Konac/driver2.java
import java.util.Scanner;
import java.io.PrintWriter;


/**
 * @author Hamza KONAÇ
 */
public class driver2
{
    /**
     * 
     * @param args main
     */
    public static void main(String args[])
    {
        String temp;
        String board_size;
        Tetris temp_obj;
        boolean control=true;
        Scanner scanner=new Scanner(System.in);
        while (true)
        {
            System.out.print("1:PLAY GAME\n");
            System.out.print("2:EXIT GAME\n\n");
            temp=scanner.nextLine();
            temp=temp.replaceAll("\n", "");
            System.out.println(temp);

            // getline(cin,temp);
            if(temp.equals("1"))
            {
                while(true)
                {
                    System.out.println("Enter Game Board Size (sample:15x15) ");
                    temp=scanner.nextLine();
                    temp=temp.replaceAll("\n", "");
                    try {
                        temp_obj=new  Tetris(temp);
                        if(temp_obj.get_game_board().size()!=0)
                            break;
                    
                    } catch (Exception e) {
                        System.out.println("Object couldnt create because of wrong size input\n");
                        // TODO: handle exception
                    }
                }
                    int move_count,move_direct,direct_count,rotate_direct;
                    while (true)
                    {
                        System.out.println("\033c");
                        temp_obj.draw();
                        control=true;
                        System.out.print("1.Enter Tetromino Name\n");
                        System.out.print("Q:Save file and quit\n");
                        temp=scanner.nextLine();
                        temp=temp.replaceAll("\n", "");
                        // getline(cin,temp);
                        if(temp.equals("I")||temp.equals("i"))
                            temp_obj.Play_game(shapes.I);
                        else if(temp.equals("o")||temp.equals("O"))
                            temp_obj.Play_game(shapes.O);
                        else if(temp.equals("l")||temp.equals("L"))
                            temp_obj.Play_game(shapes.L);
                        else if(temp.equals("s")||temp.equals("S"))
                            temp_obj.Play_game(shapes.S);
                        else if(temp.equals("t")||temp.equals("T"))
                            temp_obj.Play_game(shapes.T);
                        else if(temp.equals("z")||temp.equals("Z"))
                            temp_obj.Play_game(shapes.Z);
                        else if(temp.equals("J")||temp.equals("j"))
                            temp_obj.Play_game(shapes.J);
                        else if(temp.equals("q")||temp.equals("Q"))
                        {
                            System.out.print("ENTER FILE NAME...(extension of file have to be txt and file name must contain only letters");
                            temp=scanner.nextLine();
                            temp=temp.replaceAll("\n", "");
                            // getline(cin,temp);
                
                            if(temp_obj.control_file_name(temp))   
                            {
                                System.out.print("TRUE FILE NAMEEEE");
                                //  String file_name("output");
                                //     file_name=file_name+to_String(TetrisVector.total_object)+".txt";
                                PrintWriter file = null;

                                try
                                {
                                    file = new PrintWriter(temp, "UTF-8");   
                                }
                                catch (Exception ex)  
                                {
                                    System.out.printf("The text cannot open!\n");
                                }
                                temp_obj.writeToFile(file);
                                file.close();
                            }
                            else    
                                System.out.print("FALSE FILE NAMEEE");
                            break;
                        }
                        else
                        {
                            System.out.print("Invalid input");
                            control=false;

                        }
                        if(control)
                        {

                            String tetromino_last= new String(temp);
                            // temp_obj.Play_game(shapes.);

                            System.out.print("Choose Rotate Direction \n");
                            System.out.println("1-LEFT\n");
                            System.out.println("2-RIGHT\n");
                            System.out.println("3-NONE");
                            while(true)
                            {
                                // getline(cin,temp);
                                temp=scanner.nextLine();
                                temp=temp.replaceAll("\n", "");
                                if((temp.charAt(0)<'1'||temp.charAt(0)>'3')||temp.length()>1||temp.length()==0) 
                                    System.out.print("Invalid temp.try Again!!\n");
                                else
                                    break;
                            }
                            rotate_direct=Integer.parseInt(temp);        
                            while(true)
                            {
                                System.out.print("Enter count of rotate  direction number\n");
                                // getline(cin,temp);
                                temp=scanner.nextLine();
                                temp=temp.replaceAll("\n", "");
                                if((temp.charAt(0)<'1'||temp.charAt(0)>'5')||temp.length()>1||temp.length()==0) 
                                {
                                    System.out.print("Invalid temp.try Again!!");
                                    //return Play_game();
                                }
                                else 
                                    break;
                            }
                        
                            direct_count=Integer.parseInt(temp);
                            System.out.print("Choose Move Direction \n");
                            System.out.println("1-LEFT\n");
                            System.out.println("2-RIGHT\n");
                            System.out.println("3-NONE\n\n");
                            
                            while (true)
                            {
                                // getline(cin,temp);
                                temp=scanner.nextLine();
                                temp=temp.replaceAll("\n", "");

                                if((temp.charAt(0)<'1'||temp.charAt(0)>'3')||temp.length()>1||temp.length()==0) 
                                {
                                    System.out.print("Invalid temp.try Again!!\n");
                                }
                                else
                                    break;
                            }
                            
                            move_direct=Integer.parseInt(temp);
                            while (true)
                            {
                                System.out.print("Enter count of move  direction number\n");
                                // getline(cin,temp);
                                temp=scanner.nextLine();
                                temp=temp.replaceAll("\n", "");
                                if(Tetris.is_number(temp)) 
                                {
                                    move_count=Integer.parseInt(temp);

                                    break;
                                }
                                else
                                    System.out.print("Invalid temp.try Again!!\n");
                            }
                            temp_obj.set_variables(move_count,move_direct,direct_count,rotate_direct);
                            temp_obj.Play_game(temp_obj.get_objec_of_tetrominos().get_shape_of_obj());
                            temp_obj.animate();
                            System.out.print(temp_obj.lastMove(tetromino_last));
                            System.out.print("\nTotal move number of Game is ");
                            System.out.print(temp_obj.numberOfMoves());
                            System.out.println("\n");;

                        }
                        
                            /* code */
                    }
                    
                

            }
            else if(temp.equals("2"))
            {
                System.out.print("GAME'S HAS FINISHED\n");
                    break;
            }
        }
        scanner.close();
    }
}
